src/modules/hip/handlehip.cpp

```diff
@@ -245,6 +245,8 @@ struct HandleImpl
 
         hipMalloc(&(this->initHandle->mem.mgpu.rgbArr.rgbmem), sizeof(RpptRGB) * this->nBatchSize);
         hipMalloc(&(this->initHandle->mem.mgpu.maskArr.floatmem), sizeof(Rpp32f) * 8294400);    // 3840 x 2160
+        hipMalloc(&(this->initHandle->mem.mgpu.uintReductionArr.uintmem), sizeof(Rpp32u) * 8294400);
```

Comment:
Multiple 4K mem allocations aren't needed, lets just use the same maskArr, but reinterpret it as Rpp32u or Rpp32s pointer in the hip_exec function.

Reply:
Done